twingate/internal/client/connector.go

```diff
@@ -10,57 +10,92 @@ import (
 
 const connectorResourceName = "connector"
 
-func (client *Client) CreateConnector(ctx context.Context, remoteNetworkID, connectorName string) (*model.Connector, error) {
-	if remoteNetworkID == "" {
+func (client *Client) CreateConnector(ctx context.Context, input *model.Connector) (*model.Connector, error) {
+	if input == nil || input.NetworkID == "" {
 		return nil, NewAPIError(ErrGraphqlNetworkIDIsEmpty, "create", connectorResourceName)
 	}
 
+	if input.StatusUpdatesEnabled != nil {
+		return client.createConnectorWithNotificationStatus(ctx, input)
```

Comment:
This doesn't seem like the right solution...  what if we had more optional variables?   we'll need to create a query struct and function for every possible permutation?

Reply:
we can set default value like this, 
but the downside that if you change default value on the backend side, we would need to update client as well

```
        statusUpdatesEnabled := true	// default value
	if input.StatusUpdatesEnabled != nil {
		statusUpdatesEnabled = *input.StatusUpdatesEnabled
	}

	variables := newVars(
		gqlID(input.NetworkID, "remoteNetworkId"),
		gqlNullable(input.Name, "connectorName"),
		gqlVar(statusUpdatesEnabled, "hasStatusNotificationsEnabled"),
	)
```

Reply:
in the issue there is one point listed

```
If the argument is not specified, the default value must match the API behavior 

(which is true in this case, but in principle could be changed at the API level in the future).
```

Reply:
@ekampf it's fixed now, thank you for the backend support :)